Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,127 @@
+
+In this guide we will learn about:  
+ 
+- Controlling egress access to services outside the cluster using DNS-based policies
+- Using patterns (or wildcards) to whitelist a subset of DNS domains
+- Combining DNS and port rules for restricting access to external service  
+
+Step 3: Create Example App
+==========================
+
+In line with our Star Wars theme examples, we will use simple scenario with two pods, ``mediabot`` and ``spaceship``. The ``mediabot`` pod needs access to Twitter API services for managing the empire's tweets. However, the ``spaceship`` pods shouldn't have any external service access for obvious reasons. 
+
+.. parsed-literal::
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/minikube/dns-sw-app.yaml
+    $ kubectl get po
+    NAME                             READY   STATUS    RESTARTS   AGE
+    pod/mediabot                     1/1     Running   0          14s
+    pod/spaceship                    1/1     Running   0          14s  
+
+Step 4: Apply DNS Egress Policy
+===============================
+
+
+The following Cilium network policy allows **only** ``mediabot`` pods to access ``api.twitter.com``. 
+
+.. literalinclude:: ../../examples/policies/getting-started/dns-matchname.yaml
+
+Let's take a closer look at the policy: 
+
+* The first egress section uses ``toFQDNs: matchName`` specification to allow egress to ``api.twitter.com``. The destination DNS should match exactly the name specified in the rule. The ``endpointSelector`` allows only pods with labels ``class: mediabot`` to have the egress access.
+* The second egress section allows all pods in the ``default`` namespace to access ``kube-dns`` service. Note that ``rules: dns`` instructs Cilium to inspect and allow DNS lookups matching specified patterns. In this case, inspect and allow all DNS queries. 
+
+Let's apply the policy:
+
+.. parsed-literal::
+
+  $ kubectl create -f \ |SCM_WEB|\/examples/policies/getting-started/dns-matchname.yaml
+
+Testing the policy, we see that ``mediabot`` has access to ``api.twitter.com`` but doesn't have access to any other external service, for e.g., ``help.twitter.com``. And ``spaceship`` pod doesn't have access to twitter. 
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+    ...
+
+    $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+    ^C 
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
```

Comment:
This is *super hard* to understand. The DNS redirect policy matching on all pods has put spaceship into default deny at egress but it's absolutely random as somebody looking at the DNS redirect rule would not expect that it is also used to put spaceship into default deny to fulfill a requirement of spaceship not reaching out to the outside world.

I'm not sure we need spaceship at all. If you want to provide an example how to *restrict* access to the outside world entirely then show how to write a policy with cluster entity to only allow internal communication or show how to load a default deny policy at egress.

Reply:
Ah I see. I was a bit confused on what rule causes the "spaceship" pod to default egress deny since I didn't have any endpoint selecting it... well other than the dns egress clause :-| I will address this.

Reply:
done